utilFunctions/getContributor.js

```diff
@@ -2,8 +2,10 @@
 // return the contributor name.
 
 const getContributor = source =>
-  source.mainEntity[0]["provider"].filter(
-    ref => ref["disambiguationDescription"] == "contributing institution"
-  )[0]["name"];
+  source.mainEntity[0]["provider"]
+    ? source.mainEntity[0]["provider"].filter(
+        ref => ref["disambiguationDescription"] == "contributing institution"
+      )[0]["name"]
+    : null;
```

Comment:
Do we need to be even safer here?  
What if:

```
source.mainEntity[0]["provider"].filter(
    ref => ref["disambiguationDescription"] == "contributing institution"    
)[0]
``` 
is null?  Then when we try to call `["name"]`, it will blow up.

Maybe it would be better to put the whole thing in a try-fail block so that if the parsing fails at any point, we have a safeguard?

Reply:
added